File: Chapter04/sum.py
def sum(list:list):
    if list==[]:
        return 0
    return list[0]+sum(list[1:])

def count(list:list):
    if list==[]:
        return 0
    return 1+count(list[1:])

def max(list:list):
    if(len(list)==0):
        return None
    if(len(list)==1):
        return list[0]
    else:
        sub_max=max(list[1:])
        return list[0] if list[0]>sub_max else sub_max             

sum=sum([1,2,3,4,5])
count=count([1,2,3,4,5])
print("sum:{0},count:{1}".format(sum,count))

File: Chapter04/sum.js
const sum=(args)=>{
    if(args.length===0){
        return 0;
    }
    return args[0]+sum(args.slice(1))
}
const count=(args)=>{
    if(args.length===0){
        return 0;
    }
    return 1+count(args.slice(1));
}

const max=(args)=>{
    if(args.length===0){
        return;
    }
    if(args.length===1)
        return args[0]
    else{
        let sub_max=max(args.slice(1));
        return args[0]>sub_max?args[0]:sub_max;
    }    
}

let ArraySum=sum([1,2,3,4,5]),
    ArrayCount=count([1,2,4,5,6]),
    maxValue=max([1,5,2,3,6]);
console.log(ArraySum,ArrayCount,maxValue);